api/src/main/java/org/openmrs/module/appointments/service/impl/AppointmentsServiceImpl.java

```diff
@@ -120,6 +122,24 @@ private boolean isCurrentUserSamePersonAsOneOfTheAppointmentProviders(Set<Appoin
                 .anyMatch(provider -> provider.getProvider().getPerson().
                         equals(Context.getAuthenticatedUser().getPerson()));
     }
+    @Transactional
+    @Override
+    public Object sendAppointmentReminderSMS(Appointment appointment) {
+        String givenName = appointment.getPatient().getGivenName();
+        String familyName = appointment.getPatient().getFamilyName();
+        String patientID = appointment.getPatient().getPatientIdentifier().getIdentifier();
+        Date date = appointment.getStartDateTime();
+        String service = appointment.getService().getName();
+        List<AppointmentProvider> appointmentProviderList =new ArrayList<>(appointment.getProviders());
+        List<String> providers=new ArrayList<>();
+        for (AppointmentProvider appointmentProvider: appointmentProviderList) {
+            providers.add(appointmentProvider.getProvider().getName());
+        }
+        Location location=appointment.getLocation();
+        String message = smsService.getAppointmentMessage(givenName, familyName, patientID, date, service, providers,location);
+        PersonAttribute phoneNumber = appointment.getPatient().getAttribute("phoneNumber");
```

Comment:
Shall we check for phoneNumber first. If that is null, the rest of the code is of no use, since no message will be sent.

Reply:
You can also consider adding a log message somewhere to say, since no mobile number found for patient, therefore didn't send SMS.

Reply:
Sure will do that